crates/noirc_driver/src/lib.rs

```diff
@@ -333,37 +351,45 @@ impl Driver {
         &self,
         options: &CompileOptions,
         main_function: FuncId,
+        np_language: Language,
+        is_opcode_supported: &impl Fn(&Opcode) -> bool,
     ) -> Result<CompiledProgram, FileDiagnostic> {
         let program = monomorphize(main_function, &self.context.def_interner);
 
-        let np_language = self.language.clone();
-
-        let circuit_abi = if options.experimental_ssa {
-            experimental_create_circuit(
-                program,
-                np_language,
-                &self.is_opcode_supported,
-                options.show_ssa,
-                options.show_output,
-            )
+        let (circuit, abi) = if options.experimental_ssa {
+            let (circuit, abi) =
+                experimental_create_circuit(program, options.show_ssa, options.show_output)?;
+
+            let optimized_circuit = {
+                // TODO: Is `abi.field_count` and `circuit.current_witness_index` the same?
+                // If so, the optimization pass can be deduplicated
+                let abi_len = abi.field_count();
+
+                let simplifier = CircuitSimplifier::new(abi_len);
```

Comment:
This breaks https://github.com/noir-lang/noir/pull/1333/ as it depends on the `CircuitSimplifier` being populated as part of acir gen. This is unavoidable if we want to maintain separation between Noir ACIR gen and `acvm::compile::compile` however.

Reply:
Yep. We should make this separation so it can be architected correctly before landing. Teasing it apart afterwards will be a nightmare.